lib/elastomer/client/bulk.rb

```diff
@@ -43,6 +43,48 @@ def bulk( body = nil, params = nil )
       end
     end
 
+    # Stream bulk actions from an Enumerator.
+    #
+    # Examples
+    #
+    #   ops = [
+    #     Bulk::index(document1, :_type => 'default-type', :_id => 1),
+    #     Bulk::index(document2, params2),
+    #     Bulk::delete(params3),
+    #   ]
+    #   bulk(ops, :index => 'default-index').each do |response|
+    #     puts response
+    #   end
+    #
+    # Returns an Enumerator of responses.
+    def bulk_stream_responses(ops, params = {})
+      bulk_obj = Bulk.new(self, params)
+
+      Enumerator.new do |yielder|
+        ops.each do |op|
+          response = bulk_obj.add_to_actions *op
+          yielder.yield response unless response.nil?
+        end
+
+        response = bulk_obj.call
+        yielder.yield response unless response.nil?
+      end
+    end
+
+    # Stream bulk actions from an Enumerator.
+    #
+    # See Client#bulk_stream_responses for more information.
+    #
+    # Returns an Enumerator of items from the responses.
+    def bulk_stream_items(ops, params = {})
+      Enumerator.new do |yielder|
+        bulk_stream_responses(ops, params).each do |response|
+          response["items"].each do |item|
+            yielder.yield item
+          end
+        end
+      end
+    end
```

Comment:
My apologies, I read over your code too quickly and missed that `yield item` :bow:

I like the convenience of returning stats, and the block syntax would look almost the same as calling `each` on the `Enumerator`. I'll implement it like you showed.